### ISSUE TITLE
[Feature Request] Ability to teleport/swap windows to a location

### ISSUE THREAD

**Issue by exp2ipi** [paperwm/PaperWM#323]
One thing that I wish PaperWM could do is more easily rearrange the windows. Specifically, you would select a window, like with `alt-tab`, and instead of the layout "scrolling" to the active window, have it "teleport" in, replacing the active window (and pushing the active window to the right).

This way, for example, the user could always keep an emacs window on the left and switch between a browser and terminal window on the right.

**Comment by hedning** [comment]
Yeah, we've thought about adding something like this. Would be cool :)

An ad-hoc way to achieve something like this which I sometimes use:
- navigate to the desired window using eg. «super-arrows»
- press «t» to take the window with you
- press «Escape» to go back to the window you came from taking the window with you.

**Comment by exp2ipi** [comment]
How difficult would this be to implement? The workaround you suggested isn't very different from navigating with super-arrows, and then taking the window back to where you came from with ctrl-super-arrows. It would be really great if the user didn't have to think about where it was that they came from :)

**Comment by hedning** [comment]
I simple version might not be too much work as the [alt-tab navigator](https://github.com/paperwm/PaperWM/blob/clutter-dnd/liveAltTab.js#L102-L128) already does most of the work.

So eg. adding new actions or just detecting «ctrl» in the alt-tab navigator and then doing `space.removeWindow` and `space.addWindow` on the selected window should probably work OK ([around here](https://github.com/paperwm/PaperWM/blob/clutter-dnd/liveAltTab.js#L163-L168)).

Then there would be some more work to get it to look good. That could be some work, but when I think about it syncing the window's clone (`metaWindow.clone`) position with the one-off clone the alt-tab navigator creates might just work.

**Comment by exp2ipi** [comment]
Unfortunately, trying to add a binding to the xml schema, and copying `liveAltTab.js` to `liveAltTabTeleport.js` (with changes you suggested) only resulted in gnome crashing so hard I had to reboot my machine :)  I think I'll halt these experiments for now and hope that some more intrepid soul can tinker with this -- I do think it would be a pretty great feature to have.

**Comment by exp2ipi** [comment]
Another reason why this would be a useful feature: if gnome mouse behavior is set to anything other than "click to focus" then window focus when moving in the linear layout starts depending on whether the target window happens to fall under cursor or not. The "teleport window to you" style of switching windows would not suffer from this problem.
